Elasticsearch 2.1

ersushantsood opened the pull request:
Hi @jmaitrehenry  I have modified the changes tested and verified with elasticsearch 2.1 , can you please verify them and then we can merge these changes to master .

jmaitrehenry committed:
WIP - Add more documentation on encryption

jmaitrehenry committed:
WIP - Add more documentation on encryption

jmaitrehenry committed:
Add link to sub documentation on README

jmaitrehenry committed:
update doc

jmaitrehenry committed:
update doc

jmaitrehenry committed:
WIP

jmaitrehenry committed:
WIP

jmaitrehenry committed:
WIP - remove jdk < 8

jmaitrehenry committed:
WIP - remove jdk < 8

jmaitrehenry committed:
WIP - Fix HTTPs

jmaitrehenry committed:
WIP - Fix HTTPs

jmaitrehenry committed:
WIP - SNAPSHOT

jmaitrehenry committed:
WIP - SNAPSHOT

jmaitrehenry committed:
Merge update_doc

ersushantsood committed:
Added files via upload

ersushantsood committed:
Added files via upload

Commit:
reverted the misplaced checkins

Commit:
reverted the misplaced checkins

Commit:
changes to support Kibana 4.3 and elasticsearch 2.1

Commit:
changes to support Kibana 4.3 and elasticsearch 2.1

Commit:
README header changes

Commit:
README header changes

ersushantsood committed:
Added files via upload

ersushantsood committed:
Added files via upload

jmaitrehenry commented:
Hi @ersushantsood,
We have 2 failing tests, once those will be fixed, I will merge without problem.
Thanks for your works!

ersushantsood commented:
Hi jmaitrehenry , I have a suggestion can we keep different branches for different Elasticsearch versions as there is no backward compatibility of the plugin with older versions

jmaitrehenry commented:
I think it’s a good idea to keep a branch for ES1.7 but we can keep only one branch for 2.x?
And the latest elasticsearch version should be available in master I think.
What do you think?

> On Apr 22, 2016, at 11:50 AM, Sushant Sood <notifications@github.com> wrote:
> 
> Hi jmaitrehenry , I have a suggestion can we keep different branches for different Elasticsearch versions as there is no backward compatibility of the plugin with older versions
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub <https://github.com/petalmd/armor/pull/15#issuecomment-213485326>

ersushantsood commented:
Yap that would be ideal .We need to start working toward looking at supporting 2.2 and 2.3 versions also.

jehuty0shift committed:
adding support for Elasticsearch 2.3

(+ Dependency Updates and removal of deprecated functions in ConfigurableSearchContextCallback).

jehuty0shift committed:
adding support for Elasticsearch 2.3

(+ Dependency Updates and removal of deprecated functions in ConfigurableSearchContextCallback).

jehuty0shift committed:
adding support for Elasticesearch 2.3.3

jehuty0shift committed:
adding support for Elasticesearch 2.3.3

jehuty0shift committed:
first fully working version of A.R.M.O.R for Elasticsearch 2.3.3 with DLS and FLS filters

jehuty0shift committed:
first fully working version of A.R.M.O.R for Elasticsearch 2.3.3 with DLS and FLS filters

jehuty0shift committed:
fix copyright and author

jehuty0shift committed:
fix copyright and author

jehuty0shift committed:
not using hostname for SPNegoSchemeFactory in Tests

jehuty0shift committed:
not using hostname for SPNegoSchemeFactory in Tests

jmaitrehenry committed:
Merge pull request #16 from jehuty0shift/elasticsearch_2.3

Elasticsearch 2.3

jmaitrehenry committed:
Merge branch 'elasticsearch_2.1' of github.com:petalmd/armor into elasticsearch_2.1

jmaitrehenry commented:
@jehuty0shift & @ersushantsood I just rebase with master. If the next build is green, I will merge into master

jmaitrehenry commented on src/main/java/com/petalmd/armor/filter/AbstractActionFilter.java:
```diff
@@ -50,41 +50,61 @@
         log.debug("Context {}", request.getContext());
         log.debug("Headers {}", request.getHeaders());
 
-        if (settings.getAsBoolean(ConfigConstants.ARMOR_ALLOW_CLUSTER_MONITOR, true) && action.startsWith("cluster:monitor/")) {
```
This change break testClusterMonitorDisable

jmaitrehenry commented on src/main/java/com/petalmd/armor/filter/ArmorActionFilter.java:
```diff
@@ -125,11 +126,12 @@ private void copyContextToHeader(final ActionRequest request) {
+    private void apply0(Task task, final String action, final ActionRequest request, final ActionListener listener, final ActionFilterChain chain)
             throws Exception {
-
-        if (settings.getAsBoolean(ConfigConstants.ARMOR_ALLOW_CLUSTER_MONITOR, true) && action.startsWith("cluster:monitor/")) {
```
And this too

jehuty0shift commented:
Yep since i introduced these changes. I can fix the merge if you want. The idea behind this option was to make it possible to allow only the monitor actions that Kibana needs. Since Kibana need more than monitors action now, i could maybe rewrite it into allow_kibana_actions ? (that will allows all monitor and kibana actions needed by Kibana).

jmaitrehenry commented:
Sound good for me.
> On Jun 17, 2016, at 8:49 AM, jehuty0shift <notifications@github.com> wrote:
> 
> Yep since i introduced these changes. I can fix the merge if you want. The idea behind this option was to make it possible to allow only the monitor actions that Kibana needs. Since Kibana need more than monitors action now, i could maybe rewrite it into allow_kibana_actions ? (that will allows all monitor and kibana actions needed by Kibana).
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/petalmd/armor/pull/15#issuecomment-226760314>, or mute the thread <https://github.com/notifications/unsubscribe/AAkOmvOzJ05RpzClMeIGlZ5yR7XJpIITks5qMpfggaJpZM4INamM>.
>
